### PR TITLE
Load large set of data

### PR DESCRIPTION
**Pull request by Prakriti-nith**
This PR runs in parallel with [daru-view#104](https://github.com/SciRuby/daru-view/pull/104).

### CODE REVIEW

**Comment by Prakriti-nith** [comment]
Now, DataTables can work with a large dataset too ([~5000000](https://datatables.net/extensions/scroller/examples/initialisation/server-side_processing.html), above it may cause problems). This much of data can not loaded using client-side processing as browser hangs. Here are the screenshots for the implemented code:

Including this in controller:
datatable_ajax_controller

We can see that it is generating the datatable for 2000000 rows.
datatable_ajax

**Comment by Prakriti-nith** [comment]
I have kept client-side processing for below 50000 rows and server-side above that. It is because, in server-side processing, the searching option is not working properly.

**Comment by Prakriti-nith** [comment]
I have updated the API, done some refactoring, added docs and tests along with the feature to load large set of data(above 50000 rows). For small dataset, client-side processing is used with data being provided as js array rather than HTML table (HTML table supports only 5000 rows).

**Comment by Prakriti-nith** [comment]
Rails examples added in this PR screenshots:
rails1
rails2
rails3
rails4

**Comment by Prakriti-nith** [comment]
@Shekharrajak  can you please review this PR?

File: lib/daru/data_tables/generate_js/param_helpers.rb [hunk 1]
```diff
@@ -50,6 +52,8 @@ def typecast(value, type=nil)
           value
         end
       end
-    end # module ParamHelpers end
+      # rubocop:enable Metrics/PerceivedComplexity, Metrics/MethodLength
```
Comment (Shekharrajak) [inline lib/daru/data_tables/generate_js/param_helpers.rb]: Will try to fix these in other PRs

**Comment by Shekharrajak** [comment]
Thanks!

**Comment by Prakriti-nith** [comment]
closes #19

**Pull request merged**